CMakeLists.txt

```
@@ -48,19 +48,132 @@ if(POLICY CMP0135)
   cmake_policy(SET CMP0135 NEW)
 endif()
 
+if(WIN32)
```

This should avoid reconfiguration when the CMAKE_TOOLCHAIN_FILE still exists 
```suggestion
if(WIN32 AND NOT EXISTS "${CMAKE_TOOLCHAIN_FILE}")
```